### DISCUSSION TITLE
Seperate between relation and field

### DISCUSSION THREAD

**Discussion by erlanggadewa** [omar-dulaimi/prisma-class-validator-generator#3] [Ideas]
**There is an option to seperate between the relationship and the field??**

I want to seperate StudentProfile **relationship**  from the model (ApproveRequest.model.ts) to become ApproveRequest.model.ts and ApproveRequestRelationship.model.ts. Because your program still make the relationship field into one files

**Example prisma schema case** 👇

```
model ApproveRequest {
  id                String  @id @unique @default(uuid()) @map("id")
  identifyNumber    String  @map("identify_number") @db.VarChar(100)
  requestDate       String  @map("request_date") @db.VarChar(50)

  StudentProfile StudentProfile? @relation(fields: [identifyNumber], references: [identifyNumber])

}
```

My expected 2 files output 👇

**ApproveRequest.model.ts**
```
export class ApproveRequest {
  @IsDefined()
  @IsString()
  id!: string;

  @IsDefined()
  @IsString()
  identifyNumber!: string;

}

```

and 

**ApproveRequestRelationship.model.ts**

```
export class ApproveRequestRelationship {
  @IsOptional()
  StudentProfile?: StudentProfile;
}
```

**Comment by omar-dulaimi** [comment]
Hey @ErlanggaDewa 

I'm not sure if I fully understand what you want to achieve. Could you tell me in more details?
What use cases does this help you with?

**Comment by erlanggadewa** [comment]
@omar-dulaimi i already update the question to make simplify, i hope you can understand what i mean. Thank you

**Reply by omar-dulaimi** [reply]
I understand the code, I do. My problem is that I don't know why you would need such a change. Which is why I asked you about the use cases. How this separation would help you when building X. Like, what is the scenario?

**Reply by erlanggadewa** [reply]
I always extends my dto to your output files model to make easier addapt when prisma schema change rapidly, but sometimes I don't need the relationship attribute to include in my dto files for create/update etc. 

When iam delete the unnecessary attribute, it will take a long time

**Reply by omar-dulaimi** [reply]
Does [not generating relation fields] fix your problem? Or do you still need them in other cases?

**Reply by erlanggadewa** [reply]
it's help me, or you can make some option like : 
1. seperate relation
2. relation and and other field become 1 files (like your program now)
3. not generating relation field

btw, thank you very much for your help sir

**Comment by omar-dulaimi** [comment]
Created feature issue here: https://github.com/omar-dulaimi/prisma-class-validator-generator/issues/5
